scripts/validate_rst_title_capitalization.py

```diff
@@ -0,0 +1,394 @@
+#!/usr/bin/env python
+
+"""
+Author: tonywu1999, Date Edited: 01/17/2020
+
+Python script for collecting the titles in the rst files and validating
+if they follow the capitalization convention.  Prints the titles that do not
+follow the convention. Particularly used for .rst files in the doc/source folder
+
+NOTE: Run from the root directory of pandas repository
+
+Examples:
+./scripts/validate_rst_title_capitalization.py doc/source/development/contributing.rst
+./scripts/validate_rst_title_capitalization.py doc/source/
+
+Files that cannot be validated: (code crashes when validating for some reason)
+doc/source/user_guide/io.rst
+doc/source/whatsnew/v0.17.1.rst
+
+Reference: doctree elements
+http://epydoc.sourceforge.net/docutils/public/docutils.nodes.Element-class.html
+
+"""
+
+import argparse
+import sys
+from docutils.parsers.rst import Parser
+import docutils
+from docutils import nodes
+import re
+import os
+from os import walk
+from typing import Generator, List
+
+
+class suppress_stdout_stderr:
+    '''
+    Code source:
+    https://stackoverflow.com/questions/11130156/suppress-stdout-stderr-print-from-python-functions
+
+    A context manager for doing a "deep suppression" of stdout and stderr in
+    Python, i.e. will suppress all print, even if the print originates in a
+    compiled C/Fortran sub-function.
+    This will not suppress raised exceptions, since exceptions are printed
+    to stderr just before a script exits, and after the context manager has
+    exited (at least, I think that is why it lets exceptions through).
+
+    This code is needed to suppress output from the parser method
+    because the parser method prints to stdout when encountering Sphinx
+    references, as it cannot parse those at this moment.
+
+    '''
+    def __init__(self):
+        # Open a pair of null files
```

Comment:
Better avoid the comments where you explain the code, python code is very readable. Not the init, but for the other functions of you add a comment explaining what's their goal, use a docstring instead of a comment with a hash